decidim-core/app/helpers/decidim/check_boxes_tree_helper.rb

```diff
@@ -87,91 +88,122 @@ def filter_scopes_values
+                     TreePoint.new("", filter_text_for(t("decidim.proposals.application_helper.filter_scope_values.all")))
+                   end
+      TreeNode.new(
+        root_point,
+        array.map { |values| TreePoint.new(*values) }
+      )
+    end
+
+    def flat_filter_values(*types, **options)
+      scope = options[:scope]
+      types.map do |type|
+        [type, filter_text_for(t(type, scope:))]
+      end
+    end
+
+    def filter_text_for(translation)
+      content_tag(:span, translation).html_safe
+    end
+
+    private
+
     def filter_scopes_values_from_parent(scope)
       scopes_values = []
       scope.children.each do |child|
         unless child.children
-          scopes_values << TreePoint.new(child.id.to_s, scope_filter_text(translated_attribute(child.name, current_participatory_space.organization)))
+          scopes_values << TreePoint.new(child.id.to_s, filter_text_for(translated_attribute(child.name, current_participatory_space.organization)))
           next
         end
         scopes_values << TreeNode.new(
-          TreePoint.new(child.id.to_s, scope_filter_text(translated_attribute(child.name, current_participatory_space.organization))),
-          scope_children_to_tree(child)
+          TreePoint.new(child.id.to_s, filter_text_for(translated_attribute(child.name, current_participatory_space.organization))),
+          scope_children_to_tree(child, current_participatory_space)
         )
       end
 
       filter_tree_from(scopes_values)
     end
 
-    def filter_scopes_values_from(scopes)
+    def filter_scopes_values_from(scopes, participatory_space = nil)
+      organization = participatory_space&.organization || current_organization
+
       scopes_values = scopes.compact.flat_map do |scope|
         TreeNode.new(
-          TreePoint.new(scope.id.to_s, scope_filter_text(translated_attribute(scope.name, current_participatory_space.organization))),
-          scope_children_to_tree(scope)
+          TreePoint.new(scope.id.to_s, filter_text_for(translated_attribute(scope.name, organization))),
+          scope_children_to_tree(scope, participatory_space)
         )
       end
 
-      scopes_values.prepend(TreePoint.new("global", scope_filter_text(t("decidim.scopes.global")))) if current_participatory_space.scope.blank?
+      scopes_values.prepend(TreePoint.new("global", filter_text_for(t("decidim.scopes.global")))) if participatory_space&.scope.blank?
 
       filter_tree_from(scopes_values)
     end
 
-    def scope_children_to_tree(scope)
-      return if scope.scope_type && scope.scope_type == current_participatory_space.try(:scope_type_max_depth)
+    def scope_children_to_tree(scope, participatory_space = nil)
+      return if participatory_space.present? && scope.scope_type && scope.scope_type == current_participatory_space.try(:scope_type_max_depth)
       return unless scope.children.any?
 
+      organization = participatory_space&.organization || current_organization
```

Comment:
In which scenarios you are watching a participatory space from a different organization?

Reply:
In this case I kept the participatory_space organization as first option to get the organization because the method can receive a participatory space argument. In this way the method does not require to get the organization from the environment and the arguments provided are enough to generate the output

But taking a look to the use of the organization in this method, it's only passed as optional argument to translated_attribute, and the method uses current_organization if not provided as argument, so we can simplify the method ignoring completely the organization.

Reply:
OK to ignore it and simplify